### PR TITLE
fix: Unused State Definitions removed

### LINKED ISSUE #7943: (JS-0458) Definitions for unused states detected
## Description
You should avoid defining states that are not being used because unused states might affect while refactoring the codebase and introduces unnecessary operations by ReactJS. It is recommended to plan the behavior and operation of the component.

## Occurrences
There are 9 occurrences of this issue in the repository.

See all occurrences on DeepSource &rarr; [deepsource.io/gh/appsmithorg/appsmith/issue/JS-0458/occurrences/](https://deepsource.io/gh/appsmithorg/appsmith/issue/JS-0458/occurrences/)

### PR DESCRIPTION
**Pull request by Hard-Coder05**
## Description

Removed all the unused states as mentioned in the Issue.

Fixes #7943 

## Type of change

> Please delete options that are not relevant.

- Bug fix (non-breaking change which fixes an issue)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by mohanarpit — APPROVED** [review]

**Comment by mohanarpit** [comment]
@Hard-Coder05 The client code base isn't compiling. Would request you to please check our contribution guide & how to setup & build the client on your local machine. Please fix this.

**Comment by Hard-Coder05** [comment]
> @Hard-Coder05 The client code base isn't compiling. Would request you to please check our contribution guide & how to setup & build the client on your local machine. Please fix this.

Yeah sure! I am working on it!

File: app/client/src/components/editorComponents/CodeEditor/index.tsx [hunk 1]
```diff
@@ -171,13 +171,7 @@ class CodeEditor extends Component<Props, State> {
   updateLintingCallback: UpdateLintingCallback | undefined;
   private editorWrapperRef = React.createRef<HTMLDivElement>();
   constructor(props: Props) {
-    super(props);
-    this.state = {
-      isFocused: false,
-      isOpened: false,
-      autoCompleteVisible: false,
```
Comment (Rishabh-Rathod) [inline app/client/src/components/editorComponents/CodeEditor/index.tsx:178]: here, only `autoCompleteVisible` is the unused state to check that you can search in the same file if it is being used somewhere else or not. 
```
isFocused: false,
isOpened: false,
hinterOpen: false,
```
are used state. @Hard-Coder05 
Can you check the same for other files also, where changes are made.

**Review by Rishabh-Rathod — CHANGES_REQUESTED** [review]

File: app/client/src/components/editorComponents/CodeEditor/index.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (Hard-Coder05) [reply app/client/src/components/editorComponents/CodeEditor/index.tsx:178]: @Rishabh-Rathod sure!

**Comment by Hard-Coder05** [comment]
@Rishabh-Rathod You can review this PR now!

File: app/client/src/components/formControls/FilePickerControl.tsx [hunk 2]
```diff
@@ -49,10 +49,6 @@ class FieldFileInput extends React.Component<Props, FieldFileInputState> {
   constructor(props: Props) {
     super(props);
     this.refreshUppy();
-
-    this.state = {
-      text: "Select file to upload",
```
Comment (Rishabh-Rathod) [inline app/client/src/components/formControls/FilePickerControl.tsx:54]: Please Also remove the particular state from `FieldFileInputState`

File: app/client/src/pages/Applications/index.tsx [hunk 3]
```diff
@@ -943,11 +943,6 @@ class Applications extends Component<
 > {
   constructor(props: ApplicationProps) {
     super(props);
-
-    this.state = {
-      selectedOrgId: "",
```
Comment (Rishabh-Rathod) [inline app/client/src/pages/Applications/index.tsx:948]: Also remove the type from Applications state

**Pull request closed without merging**